Framework/src/Checker.cxx

```diff
@@ -70,120 +120,137 @@ Checker::~Checker()
-    // configuration of the database
-    mDatabase = DatabaseFactory::create(config->get<std::string>("qc.config.database.implementation"));
-    mDatabase->connect(config->getRecursiveMap("qc.config.database"));
-    LOG(INFO) << "Database that is going to be used : ";
-    LOG(INFO) << ">> Implementation : " << config->get<std::string>("qc.config.database.implementation");
-    LOG(INFO) << ">> Host : " << config->get<std::string>("qc.config.database.host");
-  } catch (
-    std::string const& e) { // we have to catch here to print the exception because the device will make it disappear
-    LOG(ERROR) << "exception : " << e;
-    throw;
+    // Init Checker Interface
+      std::unique_ptr<ConfigurationInterface> config = ConfigurationFactory::getConfiguration(mConfigurationSource);
+      for (const auto& checkerName: mCheckerNames){
+        const auto& moduleName = config->get<std::string>("qc.check."+checkerName+".moduleName");
+        loadLibrary(moduleName);
+        const std::string& className = config->get<std::string>("qc.check." + checkerName + ".className");
+        mChecks.insert(std::pair<std::string, CheckInterface*>(checkerName, getCheck(checkerName, className)));
+      }
   } catch (...) {
     std::string diagnostic = boost::current_exception_diagnostic_information();
     LOG(ERROR) << "Unexpected exception, diagnostic information follows:\n"
                << diagnostic;
     throw;
   }
-
-  // monitoring
+}
+void Checker::initPolicy(){
   try {
-    mCollector = MonitoringFactory::Get("infologger://");
+      const auto& checkerName = mCheckerNames.front();
+      std::unique_ptr<ConfigurationInterface> config = ConfigurationFactory::getConfiguration(mConfigurationSource);
+      std::vector<std::string> inputs;
+      const auto& conf = config->getRecursive("qc.check."+ checkerName);
+      //const auto& conf = config->getRecursive("qc.check."+mCheckerName);
+      for(const auto& [_key, dataSource]: conf.get_child("dataSource")){
+        (void)_key;
+        if (dataSource.get<std::string>("type") == "Task"){
+          inputs.push_back(dataSource.get_value<std::string>("name"));
+        }
+      }
+      mPolicy = std::shared_ptr<MonitorObjectPolicy>(new MonitorObjectPolicy(config->get<std::string>("qc.check."+checkerName+".policy"), inputs));
   } catch (...) {
     std::string diagnostic = boost::current_exception_diagnostic_information();
     LOG(ERROR) << "Unexpected exception, diagnostic information follows:\n"
                << diagnostic;
     throw;
   }
-  startFirstObject = system_clock::time_point::min();
-  timer.reset(1000000); // 10 s.
+
 }
 
 void Checker::run(framework::ProcessingContext& ctx)
 {
-  mLogger << "Receiving " << ctx.inputs().size() << " MonitorObjects" << AliceO2::InfoLogger::InfoLogger::endm;
+  mLogger << mCheckerNames.front() <<" Receiving " << ctx.inputs().size() << " MonitorObjects" << AliceO2::InfoLogger::InfoLogger::endm;
```

Comment:
Moving logger to foreach loop, where the TObjArray is recovered and the quantity of MO is known. This will hide the total number of MOs, but will show maybe more useful information that is number of MOs per input.

Reply:
It seems that you did not push these changes, but yes, I like the idea.